Add the accessPackageResourceEnvironment resource type (#11273)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/accesspackageresourceenvironment.md

---
title: "accessPackageResourceEnvironment resource type"
description: "An access package resource environment is a reference to the geolocation environment in which a resource is located."
author: "markwahl-msft"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# accessPackageResourceEnvironment resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> In [Azure AD Entitlement Management](entitlementmanagement-root.md), an access package resource environment is a reference to the geolocation environment in which a resource is located. This API supports operations on SharePoint Online resources only.

Reviewer: Might want to phrase the second as that it's only applicable to SharePoint Online sites.  Other resources are not geo-locatable IIRC.

Reviewer: Also, would it be useful to say that these are automatically provided as part of Azure AD Entilement Management.  (That's why there's no API to create them)

> In [Azure AD Entitlement Management](entitlementmanagement-root.md), an access package resource environment is a reference to the geolocation environment in which a resource is located. This environment is automatically provided as part of Azure AD Entilement Management. The API is only applicable to SharePoint Online sites.

Reviewer 2: Please change "SharePoint Online sites" to "Multi-Geo SharePoint Online sites"

## Methods
|Method|Return type|Description|
|:---|:---|:---|
|[List accessPackageResourceEnvironments](../api/accesspackageresourceenvironment-list.md)|[accessPackageResourceEnvironment](../resources/accesspackageresourceenvironment.md) collection|Retrieve a list of [accessPackageResourceEnvironment](../resources/accesspackageresourceenvironment.md) objects.|
|[Get accessPackageResourceEnvironment](../api/accesspackageresourceenvironment-get.md)|[accessPackageResourceEnvironment](../resources/accesspackageresourceenvironment.md)|Read the properties and relationships of an [accessPackageResourceEnvironment](../resources/accesspackageresourceenvironment.md) object.|

## Properties
|Property|Type|Description|
|:---|:---|:---|
|connectionInfo|[connectionInfo](../resources/connectioninfo.md)|Connection information of an environment used to connect to a resource. |
|createdBy|String|The display name of the user that created this object.|

> |createdDateTime|DateTimeOffset|The date and time that this object was created. <br>The DateTimeOffset type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`.|

Reviewer 3: ```suggestion
|createdDateTime|DateTimeOffset|The date and time that this object was created. <br>The DateTimeOffset type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `'2014-01-01T00:00:00Z'`.|
```

|description|String|The description of this *accessPackageResourceEnvironment* object.|
|displayName|String|The display name of this object.|
|id|String|The system-assigned unique identifier of the object.|

> |isDefaultEnvironment|Boolean|Determines whether this is default environment or not. It is set to `true` for all static origin systems like Azure AD groups and Azure AD Applications.|

Reviewer 3: ```suggestion
|isDefaultEnvironment|Boolean|Determines whether this is default environment or not. It is set to `true` for all static origin systems, such as Azure AD groups and Azure AD Applications.|
```

|modifiedBy|String|The display name of the entity that last modified this object.|

> |modifiedDateTime|DateTimeOffset|The date and time that this object was last modified. <br>The DateTimeOffset type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`. |

Reviewer 3: ```suggestion
|modifiedDateTime|DateTimeOffset|The date and time that this object was last modified. <br>The DateTimeOffset type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `'2014-01-01T00:00:00Z'`. |
```

---
title: "accessPackageResourceEnvironment resource type"
description: "An access package resource environment is a reference to the geolocation environment in which a resource is located."
author: "markwahl-msft"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# accessPackageResourceEnvironment resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> In [Azure AD entitlement management](entitlementmanagement-root.md), an access package resource environment is a reference to the geolocation environment in which a resource is located.

Reviewer 3: ```suggestion
In [Azure AD Entitlement Management](entitlementmanagement-root.md), an access package resource environment is a reference to the geolocation environment in which a resource is located.
```
